bob/src/cluster/quorum.rs

```diff
@@ -30,27 +30,34 @@ impl Quorum {
         debug!("PUT[{}] ~~~PUT LOCAL NODE FIRST~~~", key);
         let mut local_put_ok = 0_usize;
         let mut remote_ok_count = 0_usize;
-        let mut at_least = self.quorum;
+        let at_least = self.quorum;
         let mut failed_nodes = Vec::new();
         let (vdisk_id, disk_path) = self.mapper.get_operation(key);
-        if let Some(path) = disk_path {
+        let local_put_task = disk_path.map(|path| {
             debug!("disk path is present, try put local");
-            let res = put_local_node(&self.backend, key, data, vdisk_id, path).await;
-            if let Err(e) = res {
+            put_local_node(&self.backend, key, data, vdisk_id, path).boxed()
+        });
+        debug!(
+            "PUT[{}] need at least {} additional puts",
+            key,
+            at_least - 1
+        );
+        debug!("PUT[{}] ~~~PUT TO REMOTE NODES~~~", key);
+        let (mut tasks, mut errors) = self.put_remote_nodes(key, data, at_least - 1).await;
+        if let Some(task) = local_put_task {
+            let local_put = task.await;
```

Comment:
There is a possibility, that local put task would not start until this `await`. Please, check how it works in Tokio. 
Generally, it is better to use equivalent of `Task.WhenAll`

Reply:
I ran tests. I am right. To run both tasks you should use `tokio::join!` macro